services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy

```diff
@@ -855,20 +859,56 @@ class FlowCrudSpec extends HealthCheckSpecification {
             }.size() == broughtDownPorts.size() * 2
         }
 
-        and: "Make the first selected path unstable by bringing port down/up"
+        and: "First path is unstable (due to bringing port down/up)"
         // after bringing port down/up, the isl will be marked as unstable by updating the 'time_unstable' field in DB
         def islToBreak = pathHelper.getInvolvedIsls(firstPath).first()
         antiflap.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
         Wrappers.wait(WAIT_OFFSET) { assert islUtils.getIslInfo(islToBreak).get().state == FAILED }
         antiflap.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
         Wrappers.wait(WAIT_OFFSET) { assert islUtils.getIslInfo(islToBreak).get().state == DISCOVERED }
 
+        and: "Total cost of second path more preferable than the first one"
+        def involvedIslsInFirstPath = pathHelper.getInvolvedIsls(firstPath)
+        def totalCostFirstPath = involvedIslsInFirstPath.sum {
+            northbound.getLink(it).cost ?: 700
+        } + islCostWhenUnderMaintenance
+        def involvedIslsInSecondPath = pathHelper.getInvolvedIsls(secondPath)
+        def totalCostSecondPath = involvedIslsInSecondPath.sum { northbound.getLink(it).cost ?: 700 }
+        // result after performing 'if' condition: totalCostFirstPath - totalCostSecondPath = 1
+        if ((totalCostFirstPath - totalCostSecondPath) > 1) {
+            def islToUpdate = involvedIslsInSecondPath[0]
+            def currentCostOfIsl = northbound.getLink(islToUpdate).cost
+            def newCost = (totalCostFirstPath - totalCostSecondPath - 1) + currentCostOfIsl
+            database.updateIslCost(islToUpdate, newCost)
+        } else if ((totalCostFirstPath - totalCostSecondPath) == 0) {
+            def islToUpdate = involvedIslsInFirstPath[0]
+            def currentCostOfIsl = northbound.getLink(islToUpdate).cost
+            database.updateIslCost(islToUpdate, currentCostOfIsl + 1)
+        } else if ((totalCostFirstPath - totalCostSecondPath) < 0) {
+            def islToUpdate = involvedIslsInFirstPath[0]
+            def currentCostOfIsl = northbound.getLink(islToUpdate).cost
+            def newCost = (totalCostSecondPath - totalCostFirstPath + 1) + currentCostOfIsl
+            database.updateIslCost(islToUpdate, newCost)
+        }
```

Comment:
Why there are 3 conditions instead of 2?
```
        if (totalCostFirstPath - totalCostSecondPath > 0) {
            def islToUpdate = involvedIslsInSecondPath[0]
            def currentCostOfIsl = northbound.getLink(islToUpdate).cost
            def newCost = (totalCostFirstPath - totalCostSecondPath - 1) + currentCostOfIsl
            database.updateIslCost(islToUpdate, newCost)
        } else {
            def islToUpdate = involvedIslsInFirstPath[0]
            def currentCostOfIsl = northbound.getLink(islToUpdate).cost
            def newCost = (totalCostSecondPath - totalCostFirstPath + 1) + currentCostOfIsl
            database.updateIslCost(islToUpdate, newCost)
        }
```

Reply:
Also, use API update via links props instead of updating database directly

Reply:
> Also, use API update via links props instead of updating database directly

fixed